### PR TITLE
feat(parser): implement pattern matching

### PR DESCRIPTION
**Pull request by teran-joseluis**
## Description

I worked on some parser functions for **Pattern Matching**, also I added a resource which is our **EBNF** for the Yamil Language to review it during the developing process.

## Type of Change

Put an `x` in all the boxes that apply:

- [x] ✨ New Feature (non-breaking change which adds functionality)
- [ ] 🛠️ Bug Fix (non-breaking change which fixes an issue)
- [ ] ❌ Breaking Change (fix or feature that would cause existing functionality to change)
- [ ] 🧹 Code Refactor
- [ ] ✅ Build Configuration Change
- [x] 📝 Documentation
- [ ] 🗑️ Chore

## Pull Request Checklist

Put an `x` in all the boxes that apply:

- [x] My commit messages are detailed.
- [x] My code follows the code style of this project.
- [x] No existing features have been broken without good reason.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests pass.

### CODE REVIEW

File: app/resources/grammar.ebnf
Comment (JosuePrado) [inline app/resources/grammar.ebnf:1]: What is this class for, isn't it the same as the AST we have?

Comment (teran-joseluis) [reply app/resources/grammar.ebnf:1]: Actually it's, I decided to put it here for the ease of implementation of the parsers we have in the AST.

Comment (JosuePrado) [reply app/resources/grammar.ebnf:1]: It's not really necessary, but it's nice to have it in the repository.

Comment (teran-joseluis) [reply app/resources/grammar.ebnf:1]: Yeah, I added during the process of the writing the parser for pattern matching because I need clear reference of the grammar.

**Review by fabianromeroclaros — APPROVED** [review]

**Comment by teran-joseluis** [comment]
This **PR - 11** was opened again due to an observation from my partners about the quantity of commits in the previous PR related to pattern matching. Additionally, commented code was incorrectly merged into the **develop** branch due to while I was performing unit tests for my feature, I had found a bug into `parseLiteral` function so it forced me to comment those unit tests I wrote previously. That code has now been removed from the **develop** branch, and now I'm waiting for approval again of [PR - 16](https://github.com/Programming-Languages-I/yamil-language/pull/16) it to be merged correctly.

**Comment by teran-joseluis** [comment]
Additionally, I have deleted the branch that contained the errors and observations from my teammates, as we will no longer be using it.

**Comment by teran-joseluis** [comment]
Additionally, I have deleted this branch because we will no longer be using it and now the correct content is located on [**PR - 16**](https://github.com/Programming-Languages-I/yamil-language/pull/16) as I mentioned on the previous comment

**Pull request merged**
